src/Icicle/Dictionary/Parse/Prisms.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+module Icicle.Dictionary.Parse.Prisms (
+    _NTValue
+  , _NTable
+  , _VString
+  , _VInteger
+  , _VFloat
+  , _VBoolean
+  , _VDatetime
+  , _VArray
+  , key
+  ) where
+
+import           Control.Applicative
+import           Control.Lens
+
+import           Data.Bool
+import           Data.Char
+import           Data.Either
+import           Data.Function
+import           Data.Int
+import           Data.Text
+import           Data.Time
+
+import qualified Text.Parsec.Pos as Pos
+
+import           GHC.Float
+
+import           Icicle.Dictionary.Parse.Types
+
+
+_NTValue :: Prism' Node TValue
```

Comment:
can you use template haskell to generate this? (I am not suggesting you change it, I'm just curious)